A/ADOLC/build_tarballs.jl

```diff
@@ -0,0 +1,74 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+# See https://discourse.julialang.org/t/binarybuilder-jl-cant-dlopen-because-of-libopenlibm-so/108486
+# This build script needs to be run with julia 1.7
+
+# See https://github.com/JuliaLang/Pkg.jl/issues/2942
+# Once this Pkg issue is resolved, this must be removed
+uuid = Base.UUID("a83860b7-747b-57cf-bf1f-3e79990d037f")
+delete!(Pkg.Types.get_last_stdlibs(v"1.6.3"), uuid)
+
+name = "ADOLC"
+version = v"0.0.24"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/coin-or/ADOL-C.git", "b7ea10536a1e452f7dd47a90ef8c8118ce3e8432"),
+    GitSource("https://github.com/TimSiebert1/libadolccxx.git", "51c23abf9ee2e1566331b440e81503ac9e7e8f5b"),
+]
+
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/ADOL-C/
+autoreconf -fi
+./configure --enable-atrig-erf --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j${nproc}
+make install
+install_license LICENSE
+
+cd ../libadolccxx/
+cmake -DCMAKE_INSTALL_PREFIX=${prefix}\
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}\
+      -DCMAKE_BUILD_TYPE=Release\
+      -DADOLC_DIR=${prefix}\
+      -DJulia_PREFIX=${prefix}\
+      -B build .
+
+cd build
+make -j${nprocs}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+# See https://discourse.julialang.org/t/binarybuilder-jl-cant-dlopen-because-of-libopenlibm-so/108486
+# It seems we need a separate build for each Julia version
+include("../../L/libjulia/common.jl")
+
+julia_versions=VersionNumber[v"1.9", v"1.10", v"1.11"]
+
+platforms = vcat(libjulia_platforms.(julia_versions)...)
```

Comment:
https://buildkite.com/julialang/yggdrasil/builds/11150#01901264-7007-424b-88e6-eca382831201/666-1737
```
┌ Warning: /cache/build/yggy-amdci7-8/julialang/yggdrasil/A/ADOLC/build/x86_64-linux-gnu-julia_version+1.9.0/LNG4nM6x/x86_64-linux-gnu-libgfortran5-cxx11-julia_version+1.9.0/destdir/lib/libadolc_wrap.so.0.0.1 contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder.Auditor /cache/julia-buildkite-plugin/depots/e2fd9734-29d8-45cd-b0eb-59f7104f3131/packages/BinaryBuilder/fV5qK/src/auditor/compiler_abi.jl:247
```
```suggestion
platforms = vcat(libjulia_platforms.(julia_versions)...)
platforms = expand_cxxstring_abis(platforms)
```